pygame_gui/core/ui_appearance_theme.py

```diff
@@ -962,57 +970,19 @@ def _load_colour_or_gradient_from_theme(theme_colours_dictionary: Dict[str, str]
+        if colour is None:
+            if len(get_commas_outside_enclosing_glyphs(string_data)) > 0: # Was probably meant to be a gradient
+                warnings.warn("Invalid gradient: " + string_data + " for id:" + colour_id + " in theme file")
             else:
-                if len(string_data_list) == 3:
-                    # two colour gradient
-                    try:
-                        colour_1 = premul_col(pygame.Color(string_data_list[0]))
-                        colour_2 = premul_col(pygame.Color(string_data_list[1]))
-                        loaded_colour_or_gradient = ColourGradient(gradient_direction,
-                                                                   colour_1,
-                                                                   colour_2)
-                    except ValueError:
-                        warnings.warn("Invalid gradient: " + string_data +
-                                      " for id:" + colour_id + " in theme file")
-                elif len(string_data_list) == 4:
-                    # three colour gradient
-                    try:
-                        colour_1 = premul_col(pygame.Color(string_data_list[0]))
-                        colour_2 = premul_col(pygame.Color(string_data_list[1]))
-                        colour_3 = premul_col(pygame.Color(string_data_list[2]))
-                        loaded_colour_or_gradient = ColourGradient(gradient_direction,
-                                                                   colour_1,
-                                                                   colour_2,
-                                                                   colour_3)
-                    except ValueError:
-                        warnings.warn("Invalid gradient: " + string_data +
-                                      " for id:" + colour_id + " in theme file")
+                if string_data.startswith("#"):
+                    warnings.warn("Colour hex code: " + string_data + " for id:" + colour_id + " invalid in theme file")
                 else:
-                    warnings.warn("Invalid gradient: " + string_data +
-                                  " for id:" + colour_id + " in theme file")
-        else:
-            try:  # expecting a regular hex colour in string data
-                loaded_colour_or_gradient = premul_col(pygame.Color(string_data))
-            except ValueError:
-                warnings.warn("Colour hex code: " + string_data + " for id:" + colour_id +
-                              " invalid in theme file")
-
-        if loaded_colour_or_gradient is None:
-            # if the colour or gradient data is invalid, return a black default colour
-            loaded_colour_or_gradient = pygame.Color("#000000")
-
-        return loaded_colour_or_gradient
+                    warnings.warn("Colour: " + string_data + " for id:" + colour_id + " invalid in theme file" )
```

Comment:
Because this is a new warning it is going to trip up all the block in the element tests that look something like this:
```
@pytest.mark.filterwarnings("ignore:Invalid value")
@pytest.mark.filterwarnings("ignore:Colour hex code")
@pytest.mark.filterwarnings("ignore:Colour: ")
def test_rebuild_from_theme_data_bad_values(self, _init_pygame, _display_surface_return_none)
```

To make pytest stop complaining about the warning they will just need changing to something like this filtering out the new warning when we are purposefully testing if the element is still created even when we give bad values to the theming file:
```
@pytest.mark.filterwarnings("ignore:Invalid value")
@pytest.mark.filterwarnings("ignore:Colour hex code")
def test_rebuild_from_theme_data_bad_values(self, _init_pygame, _display_surface_return_none)
```

I believe there is one test like this in every element except `UIImage`, `UIProgressBar` and `UIScrollingContainer` and `UIDropdown` has two of these tests.

Reply:
I noticed that in this case "Colour: " is a really vague warning name, and if someone were to be reading the test cases, they'd really have no idea where to look for the warning error. I changed it to "Invalid Theme Colour" instead and added the warning filters for "Invalid Theme Colour" to all of the code blocks which define an "Invalid value" and "Colour hex code" warning filter as well